pkg/encryption/encryption_stores.go

```diff
@@ -20,55 +20,55 @@ type Stores struct {
 // See the cassandra.yaml file for explanations on how to set these options: https://github.com/apache/cassandra/blob/cassandra-4.0/conf/cassandra.yaml#L1091-L1183
 // +kubebuilder:object:generate=true
 type ClientEncryptionOptions struct {
-	EncryptionSettings `json:",inline"`
-	Enabled            bool `json:"enabled"`
-	Optional           bool `json:"optional,omitempty"`
+	EncryptionSettings `json:",inline" cass-config:"*:;recurse"`
+	Enabled            bool `json:"enabled" cass-config:"*:enabled;retainzero"`
+	Optional           bool `json:"optional,omitempty" cass-config:"*:optional"`
 }
 
 // See the cassandra.yaml file for explanations on how to set these options: https://github.com/apache/cassandra/blob/cassandra-4.0/conf/cassandra.yaml#L1091-L1183
 // +kubebuilder:object:generate=true
 type ServerEncryptionOptions struct {
 	// +kubebuilder:default=false
 	// +kubebuilder:validation:optional
-	Optional *bool `json:"optional,omitempty"`
+	Optional *bool `json:"optional,omitempty" cass-config:">=4.x:optional"`
 
 	// +kubebuilder:validation:Enum=none;dc;rack;all
 	// +kubebuilder:default=none
 	// +kubebuilder:validation:optional
-	InternodeEncryption string `json:"internode_encryption,omitempty"`
+	InternodeEncryption string `json:"internode_encryption,omitempty" cass-config:"*:internode_encryption"`
 
 	// +kubebuilder:default=false
 	// +kubebuilder:validation:optional
-	RequireEndpointVerification bool `json:"require_endpoint_verification,omitempty"`
+	RequireEndpointVerification bool `json:"require_endpoint_verification,omitempty" cass-config:"*:require_endpoint_verification"`
 
 	// +kubebuilder:default=false
 	// +kubebuilder:validation:optional
-	EnableLegacySslStoragePort bool `json:"enable_legacy_ssl_storage_port,omitempty"`
+	EnableLegacySslStoragePort bool `json:"enable_legacy_ssl_storage_port,omitempty" cass-config:"*:enable_legacy_ssl_storage_port"`
 
-	EncryptionSettings `json:",inline"`
+	EncryptionSettings `json:",inline" cass-config:"*:;recurse"`
 }
 
 // See the cassandra.yaml file for explanations on how to set these options: https://github.com/apache/cassandra/blob/cassandra-4.0/conf/cassandra.yaml#L1091-L1183
 // +kubebuilder:object:generate=true
 type EncryptionSettings struct {
 	// +kubebuilder:validation:optional
-	Protocol string `json:"protocol,omitempty"`
+	Protocol string `json:"protocol,omitempty" cass-config:"*:protocol"`
```

Comment:
isn't the path redundant with the json tag in case of an inline setting?
Have you considered using the json tag for the setting output name, and only specifying a path in the cass-config tag?
Not sure how much you can handle one tag using info from another, but seeing the repetition of the name in both tags feels like something could be optimized.

Reply:
We could do that, but in fact I like decoupling the two tags, even if that leads to some repetition.

With the changes in this PR:

* The `json` tag is used only for the CRD;
* The `cass-config` tag is used only for config-builder payloads.

This allows you to specify different values: one for how the field appear in the CRD and another for how it is passed to config-builder. I did it for the JVM options for example, and thus the CRD JVM options look more "homogenous" since I harmonized the option names.

Reply:
I understand the flexibility it gives, but do we really need different names for the CRD and the config-builder options?
I went back and checked, and it seems like it lead you to specialize some settings for G1 although they are used by all collectors, such as `ParallelGCThreads` or `ConcGCThreads` (see [here](https://docs.oracle.com/en/java/javase/11/gctuning/z-garbage-collector1.html) for example, but it applies to CMS as well).

I'd be keen to retain the original jvm option name throughout the chain, otherwise folks will have to look for a mapping before setting them. We would make it a smoother experience if the names were just converted to snake case, don't you agree?

Reply:
> I'd be keen to retain the original jvm option name throughout the chain [...] We would make it a smoother experience if the names were just converted to snake case

Well, if you prefer... but the names will be a mess :-)
@jsanda what would be your preference?

> it seems like it lead you to specialize some settings for G1 although they are used by all collectors, such as ParallelGCThreads or ConcGCThreads

Unfortunately those settings will only work for G1. This is because config-builder defines a conditional on the GC name for those settings to be taken into account, see [here](https://github.com/datastax/cass-config-definitions/blob/f3f5e40c780100990729abc1c62498c520c26815/resources/jvm11-server-options/cassandra/jvm11-server-options-cassandra-4.0.0.edn#L79-L90) for C* 4.0 and [here](https://github.com/datastax/cass-config-definitions/blob/7e019d117fd0324b43c072c19648276d09de9f9e/resources/jvm-options/cassandra/jvm-options-cassandra-3.11.6.edn#L222-L251) for C* 3.11. That's why I prefixed them with "gc_g1_".

Reply:
> Unfortunately those settings will only work for G1. This is because config-builder defines a conditional on the GC name for those settings to be taken into account, see [here](https://github.com/datastax/cass-config-definitions/blob/f3f5e40c780100990729abc1c62498c520c26815/resources/jvm11-server-options/cassandra/jvm11-server-options-cassandra-4.0.0.edn#L79-L90) for C* 4.0 and [here](https://github.com/datastax/cass-config-definitions/blob/7e019d117fd0324b43c072c19648276d09de9f9e/resources/jvm-options/cassandra/jvm-options-cassandra-3.11.6.edn#L222-L251) for C* 3.11. That's why I prefixed them with "gc_g1_".

😱

Reply:
I know... config-builder sucks sometimes. The only way to pass those options when using another gc is to use `additionalOptions`.

Reply:
@jsanda would you mind chiming in here if you have time?

I fear that the alignment of option names as exposed to the K8ssandra user with actual option names as recognized by both config-builder and Cassandra is a pipedream. There are already cases where the option name changes with the Cassandra version (e.g. biased locking). And with the changes from CASSANDRA-15234, the config options for C* 4.1 will change even more drastically.

Reply:
Given what we know will happen in 4.1, you can discard my comment. Having different names for both our input and output will probably be more helpful than I thought.